### PR TITLE
Change logic for the new version handling

### PR DESCRIPTION
**Pull request by ludeeus**
## Proposed change

Required frontend changes for the new version handling purposed in <https://github.com/home-assistant/supervisor/pull/2155>

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml

```

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue:
- Link to documentation pull request:

## Checklist

- [x] The code change is tested and works locally.
- [ ] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

File: hassio/src/dashboard/hassio-addons.ts [hunk 1]
```diff
@@ -52,22 +52,22 @@ class HassioAddons extends LitElement {
                           .title=${addon.name}
                           .description=${addon.description}
                           available
-                          .showTopbar=${addon.installed !== addon.version}
+                          .showTopbar=${addon.installed ||
+                          addon.update_available!}
```
Comment (bramkragten) [inline hassio/src/dashboard/hassio-addons.ts:56]: This is a different logic than before, intended?

File: hassio/src/dashboard/hassio-update.ts [hunk 2]
```diff
@@ -39,35 +40,33 @@ export class HassioUpdate extends LitElement {
...
 
-  @internalProperty() private _error?: string;
+  private _pendingUpdates = memoizeOne(
+    (
+      core?: HassioHomeAssistantInfo,
+      supervisor?: HassioSupervisorInfo,
+      os?: HassioHassOSInfo
+    ): number => {
+      return [core, supervisor, os].filter(
+        (value) => !!value && value?.version !== value?.version_latest
```
Comment (bramkragten) [inline hassio/src/dashboard/hassio-update.ts:52]: supervisor and os now have `update_available` why not use that here?

File: src/data/hassio/supervisor.ts [hunk 3]
```diff
@@ -1,19 +1,41 @@
 import { HomeAssistant, PanelInfo } from "../../types";
+import { HassioAddonInfo, HassioAddonRepository } from "./addon";
 import { hassioApiResultExtractor, HassioResponse } from "./common";
 
 export type HassioHomeAssistantInfo = any;
```
Comment (bramkragten) [inline src/data/hassio/supervisor.ts:5]: Don't we have typing for this one?

File: hassio/src/dashboard/hassio-addons.ts (hunk 1 above)
Comment (ludeeus) [reply hassio/src/dashboard/hassio-addons.ts:56]: No

File: src/data/hassio/supervisor.ts (hunk 3 above)
Comment (ludeeus) [reply src/data/hassio/supervisor.ts:5]: Now we do

**Review by bramkragten — APPROVED** [review]

**Pull request merged**